### ISSUE TITLE
Deepseek not working on existing projects

### ISSUE THREAD

**Issue by clickway** [stackblitz-labs/bolt.diy#956]
### Describe the bug

Thought initially was a problem with the api key, however after running some tests, I can see the issue coming from the existing project I'm working on. So if I'm to start a new project from scratch this I would not encounter this issue. 

---
A bit of background: Started the project initially on Bolt.new, moved to the local version of Bolt.diy, continue the work using Google Gemini API 2.0 - relatively fine, then today I tried to use the newest version of Deepseek v.3 both coder/chat and it doesn't work, getting this error when using the prompt:
![image](https://github.com/user-attachments/assets/c450c62c-7d99-464a-8ea8-4e97b58d4348)
``

### Link to the Bolt URL that caused the error

localhost app

### Steps to reproduce

1. Load a current project you are working on via Bolt.diy
2. Change the LLM to DeepSeek API
3. Type any request, or simple 'test' 
4. Error shown via the toast message (see screenshot)

### Expected behavior

to work? 

### Screen Recording / Screenshot

![deepseek_error](https://github.com/user-attachments/assets/1e381376-b9db-479a-989f-bb96970ac636)

### Platform

- OS: [macOS]
- Browser: [Chrome]
- Version: [latest]

**Comment by robsonlimadeveloper** [comment]
I have the same problem. It doesn't seem to be a key issue. In my case no prompt works using deepseek.
![image](https://github.com/user-attachments/assets/7dbef86e-9105-403c-891f-3c2e24f00807)

**Comment by tasktuner** [comment]
Same issue here

**Comment by electroheadfx** [comment]
Same issue, is it a problem with V3 update ? or maybe the api url is wrong ? `https://api.deepseek.com/beta'`

**Comment by mbenhard** [comment]
Same issue here. I think it might be due to the low input context window (64K when using their API). I might be wrong though.

Also noticed that by the default DeepSeek sets maximum output length to 4K and **max_tokens** value needs to be set, not sure how it is configured in bolt.diy

**Comment by dev1ender** [comment]
Reply @mbenhard 

Actually its configuration seems right as [ Deepseek DOC] (https://api-docs.deepseek.com/quick_start/pricing ) says output context length is 8k not 4k also the bolt.diy configuration correctly configured to 8K 
```
path: app/lib/modules/llm/providers/deepseek.ts

  staticModels: ModelInfo[] = [
    { name: 'deepseek-coder', label: 'Deepseek-Coder', provider: 'Deepseek', maxTokenAllowed: 8000 },
    { name: 'deepseek-chat', label: 'Deepseek-Chat', provider: 'Deepseek', maxTokenAllowed: 8000 },
  ];

```

Not sure is because of that

**Comment by dwlynch055** [comment]
same issue here.

**Comment by clickway** [comment]
> Reply @mbenhard
> 
> Actually its configuration seems right as [ Deepseek DOC](https://api-docs.deepseek.com/quick_start/pricing) says output context length is 8k not 4k also the bolt.diy configuration correctly configured to 8K
> 
> ```
> path: app/lib/modules/llm/providers/deepseek.ts
> 
>   staticModels: ModelInfo[] = [
>     { name: 'deepseek-coder', label: 'Deepseek-Coder', provider: 'Deepseek', maxTokenAllowed: 8000 },
>     { name: 'deepseek-chat', label: 'Deepseek-Chat', provider: 'Deepseek', maxTokenAllowed: 8000 },
>   ];
> ```
> 
> Not sure is because of that

Look it works, if you were to start a new prompt chat from scratch, and is quite fast compared to Google Gemini but no matter how you try to use it on an existing project or start new project and import a folder then it fails to do anything.

**Comment by tasktuner** [comment]
Yep same here @clickway I think thats the main issue. Not with new code

**Comment by electroheadfx** [comment]
> new prompt chat from scratch,

No worked for me from a new prompt chat from scratch

**Comment by elivonai91** [comment]
same issue here

**Comment by Mistes974** [comment]
Is there any workaround ?

**Comment by bilalinamdar** [comment]
I also purchased top up from deep and after entering the api receiving same error. not able to bypass it.

**Comment by komax74** [comment]
I am in the same situation; if I create a new Deepseek project, it works perfectly. However, I continuously receive a timeout error if I try to modify a project created on bolt.new and exported (which works perfectly in preview but I get an error with every thing I write when trying to edit it). I also tried ChatGPT 4, same issue, only Gemini works.
Please help us solve this problem as I am spending a lot of tokens on bolt.new, it would be great to continue the project.

**Comment by electroheadfx** [comment]
> I am in the same situation; if I create a new Deepseek project, it works perfectly. However, I continuously receive a timeout error if I try to modify a project created on bolt.new and exported (which works perfectly in preview but I get an error with every thing I write when trying to edit it). I also tried ChatGPT 4, same issue, only Gemini works. Please help us solve this problem as I am spending a lot of tokens on bolt.new, it would be great to continue the project.

@komax74 @kyo-1  How you do that, new Deepseek project out me the error too.

**Comment by shivansh488** [comment]
in my case it is not working in any of the above specified case

**Comment by Kingjam99** [comment]
Same issue

**Comment by 3assemm** [comment]
I am facing the same problem with all other Models except with Google. Even when using Google, after exceeding the 8k, it throws that error and try to use LLM Anthropic  

app-dev-1  |  INFO   api.chat  Reached max token limit (8000): Continuing message (1 switches left)
2025-01-05 16:27:59 app-dev-1  |  INFO   stream-text  Sending llm call to Anthropic with model claude-3-5-sonnet-latest 
![image](https://github.com/user-attachments/assets/5331880e-e541-4263-a519-cf30d670c361)

**Comment by RaphPi** [comment]
Is there a link with the #820 issue ?

**Comment by abdellatiff** [comment]
Same issue here, but i got no error message at all, i just import new project either from file or from git, basically the project was in bolt.new, but nothing happens when i right any prompt, it keeps loading forever, but it works in a new chat.

**Comment by Tmkl972** [comment]
Hello everyone, I am experiencing the same problem. I am interested in finding a solution.

**Comment by Ishannaik** [comment]
Same issue any solution yet?

**Comment by openport-cmd** [comment]
Same issue, ideas?
?

**Comment by Sergio-Daniel-Pires** [comment]
I get this Issue when I tried to run on a existing project, the request gets around ~140k/tokens, as paliative solution I deleted package-lock.json and get around 11k tokens/request.

**Comment by leex279** [comment]
This is most likely a problem with the API key you using or the provider itself. For Deepseek e.g. you have to pay credits to make the API work (its not free!).
For OpenAI you also need to pay at least 5 Euro I guess to get to Tier1 and usable API.

The main branch has extended logging for the terminal (new features). Check this out, try your prompt again and check the terminal for errors.

**Comment by abhishek4tech84** [comment]
> This is most likely a problem with the API key you using or the provider itself. For Deepseek e.g. you have to pay credits to make the API work (its not free!). For OpenAI you also need to pay at least 5 Euro I guess to get to Tier1 and usable API.
> 
> The main branch has extended logging for the terminal (new features). Check this out, try your prompt again and check the terminal for errors.

No I dont think.....The same happens for existing project of Google Gemini 2.0

**Comment by Daybayzayd** [comment]
> [@abhishek4tech84](https://github.com/abhishek4tech84) I am pretty sure it is, cause we had this topic a lot in the community. Maybe not in your special case then, but 90% its the case.
> 
> If you want to investigate your case, I would recommend opening a topic in community: https://thinktank.ottomator.ai/c/bolt-diy/bolt-diy-issues-and-troubleshooting/22

No.
I created my project on bolt.new. I downloaded boly.diy, imported my project inside, configured my API Key with Deepseek (i have paid the API Key) and there is big problem. It doesn't write entirely the content file, it doesn't modify always directly the files but write the code in the prompt, there is often errors on the code returned...

**Comment by leex279** [comment]
@Daybayzayd Issue is closed => if you want to investigate use the link to the community and open a topic for your case.

**Comment by Bigwrm** [comment]
> > [@abhishek4tech84](https://github.com/abhishek4tech84) I am pretty sure it is, cause we had this topic a lot in the community. Maybe not in your special case then, but 90% its the case.
> > If you want to investigate your case, I would recommend opening a topic in community: https://thinktank.ottomator.ai/c/bolt-diy/bolt-diy-issues-and-troubleshooting/22
> 
> No. I created my project on bolt.new. I downloaded boly.diy, imported my project inside, configured my API Key with Deepseek (i have paid the API Key) and there is big problem. It doesn't write entirely the content file, it doesn't modify always directly the files but write the code in the prompt, there is often errors on the code returned...

I'm having this same issue in the same situation.

**Comment by DANGUTVA** [comment]
same for me!

**Comment by marvin-rao** [comment]
Also running into this issue, with local bolt.diy, llama works fine with Ollama, but when using deepseek in Ollama, the chat request just doesn't finish.
